BaseCode/App/Routing/Sheet.swift

```diff
@@ -9,12 +9,32 @@ import UIKit
 
 enum Sheet {
 
-    case confirmation
-
+    case ageSelection(age: Int, selectedAge: ((_ age: Int) -> Void))
+    case avatar(avatar: Avatar?, selectedAvatar: ((Avatar?) -> Void))
+    case policy
+    
     func controller() -> UIViewController {
         switch self {
-        case .confirmation:
-            return UIViewController()
+        case let .ageSelection(age, selectedAge):
+            let viewModel = AgeSelectionViewModel()
+            viewModel.selectedRow = age - 1
+            viewModel.selectedAge = selectedAge
```

Comment:
Couple of problems here.
Prefer passing these values in init method
```swift
AgeSelectionViewModel(age: age)
```
view model should able to calculate the selected row by itself. The `Sheet` and `Route` enums are only for initialization and passing data. you should not be writing any logic here